Content.Shared/Preferences/HumanoidCharacterProfile.cs

```diff
@@ -534,9 +534,24 @@ public void EnsureValid(ICommonSession session, IDependencyCollection collection
                 name = GetName(Species, gender);
             }
 
-            var customspeciename = speciesPrototype.CustomName
-                ? FormattedMessage.RemoveMarkup(Customspeciename ?? "")[..MaxNameLength]
-                : "";
+            string customspeciename;
+            if (string.IsNullOrEmpty(Customspeciename))
+            {
+                customspeciename = "";
+            }
+            else if (Customspeciename.Length > MaxNameLength)
+            {
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength];
+            }
+            else
+            {
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename);
+            }
+
+            if (!speciesPrototype.CustomName)
+            {
+                customspeciename = "";
+            }
```

Comment:
```suggestion
            if (string.IsNullOrEmpty(Customspeciename))
                customspeciename = "";
            else if (Customspeciename.Length > MaxNameLength)
                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength];
            else
                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename);

            if (!speciesPrototype.CustomName)
                customspeciename = "";
```

Reply:
note: i just reverted what caused the issues, not made it better so this pehaps could be.